frame/babe/src/mock.rs

```diff
@@ -314,17 +314,16 @@ pub fn make_secondary_vrf_pre_digest(
 	Digest { logs: vec![log] }
 }
 
-pub fn make_vrf_output(
+pub fn make_vrf_signature_and_randomness(
 	slot: Slot,
 	pair: &sp_consensus_babe::AuthorityPair,
 ) -> (VrfSignature, Randomness) {
-	let transcript = sp_consensus_babe::make_transcript(&Babe::randomness(), slot, 0);
+	let transcript = sp_consensus_babe::make_vrf_transcript(&Babe::randomness(), slot, 0);
```

Comment:
Maybe dumb question: Why cannot we use `make_vrf_sign_data` here?
In the `client/consensus/babe/src/tests.rs` output of `make_vrf_sign_data` is pass to `vrf_sign`.
What is the difference?

Reply:
Because we use the transcript to call `make_bytes` first.

A transcript can be used to:
1. output some random bytes  (when used as a `VrfInput` for make_bytes)
2. part of the the data to be signed (when used within the `VrfSignData`)

If we require to first do the first action then is more convenient to transform the transcript into `VrfSignData` after (even though I added a method to get a reference to the transcript from `VrfSignData` as well)

Indeed in babe we can (in the future) first output the lottery score bytes, check if the output is below the threshold and then sign the corresponding `VrfSignData` only if the score is below the threshold.

Currently instead (in BABE) we first sign and then generate the vrf output bytes from the signature (yes the output bytes can be generated from the signature as well, and are the same as the bytes generated directly from transcript used to construct the `VrfSignData`).

Before this PR BABE was not having other choice because the keystore was not providing a method to make score bytes from some sort of input (in this case transcript) and thus was preventing the user to perform a "check before sign" operation.

We'll require this for keystore method sassafras

Reply:
One more question to widen my knowledge:
If I understand it correctly, `VrfInput` is just part of `VrfSignData` in case of SR25519. So theoretically `make_bytes` could be called using `VrfSignData` (and we could get rid of `sp_consensus_babe::make_vrf_transcript`).

Is this relation true (_input_ being explicit part of _signdata_) for other VRF systems?

Reply:
yes `make_bytes` can be called using `VrfSignData` but this struct contains some extra stuff not really required if all you want to do is grab some output bytes